Code2CollegeBankingProject/BankingUI/views.py

```diff
@@ -1,132 +1,248 @@
-    "email":customerbankinfo.customer_info.email
+    "firstname":customer_bank_info.first_name,
+    "lastname":customer_bank_info.last_name,
+    "email":customer_bank_info.email,
+    "money": customer_amount,
+    "transactions": date_transaction_type_amount
     }
     return render(request,'account.html', user)
 
 def changeaccountinfo(request):
-    return render(request,'changeaccountinfo.html')
+    userid = request.session['userid']
+    customer_bank_info = User.objects.get(id = userid)
+    if request.method == "POST":
+        first_name = request.POST["firstname"]
+        last_name = request.POST["lastname"]
+        email = request.POST["email"]
+        username = request.POST["username"]
+        if User.objects.filter(username = username).exists() == False or customer_bank_info.username == username:#if the username doensn't exist
+            if User.objects.filter(email = email).exists() == False or customer_bank_info.email == email:#if the email doensn't exist
+                customer_bank_info.first_name = first_name
+                customer_bank_info.last_name = last_name
+                customer_bank_info.email = email
+                customer_bank_info.username = username
+                customer_bank_info.save()
+                user = {
+                    "title":customer_bank_info.username,
+                    "response":"Your new information has been saved"
+                }
+                return render(request,'changeaccountinfo.html',user)
+            else:
+                user = {
+                    "title":customer_bank_info.username,
+                    "response":"The email is already in use"
+                }
+                return render(request,'changeaccountinfo.html',user)
+        else:
+            user = {
+                "title":customer_bank_info.username,
+                "response":"The username is already in use"
+            }
+            return render(request,'changeaccountinfo.html',user)
+    user = {
+            "title":customer_bank_info.first_name
+        }
+    return render(request,'changeaccountinfo.html',user)
 
 def changemoney(request):
     userid = request.session['userid']
-    customerbankinfo = BankInfo.objects.get(id = userid)
+    customer_bank_info = User.objects.get(id = userid)
     if request.method == "POST":
-        transactiontype = request.POST.get('transactiontype')
+        transaction_type = request.POST.get('transactiontype')
         amount = request.POST.get('amount')
-        newtransaction = Transactions(transactiontype = transactiontype, amount = amount, account = customerbankinfo)
+
+
+        customer_transactions = Transactions.objects.filter(account = customer_bank_info.id)
+        customer_amount = 0
+        for transactions in customer_transactions:
+            if transactions.transactiontype == "Deposit":
+                customer_amount = customer_amount + transactions.amount
```

Comment:
Ooh... there is an easy syntax trick that this can simplified to:

```suggestion
                customer_amount +=  transactions.amount
```
This does the exact same thing.